### PR TITLE
Add Fabrice Drouin - Attack Vectors of Lightning Network, Chaincode Residency Summer 2019

### PR DESCRIPTION
**Pull request by gijswijs**

### CODE REVIEW

File: chaincode-labs/chaincode-residency/2019-06-25-fabrice-drouin-attack-vectors-of-lightning-network.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+There's an ongoing DDos attack  against electrum servers, I don’t know if you have heard of it. They are blacklisting I think 150,000 IPs right now, but it's a battle that is really hard to fight. Something that is really easy to do with lightning but it's a bit costly is to just lock up channels. You have limitations in lightning. You can't have more than, you can’t have too many pending payments and the amounts of your pending payments is also limited, so you can just lock up channels. You can just send HTLC’s without pre-images and after a while channels will not accept more pending payments. However this is not free. So can you tell me why it's not really free to do that?
```
Comment (adamjonas) [inline chaincode-labs/chaincode-residency/2019-06-25-fabrice-drouin-attack-vectors-of-lightning-network.md:26]:
```suggestion
There's an ongoing DDos attack against electrum servers, I don’t know if you have heard of it. They are blacklisting I think 150,000 IPs right now, but it's a battle that is really hard to fight. Something that is really easy to do with lightning but it's a bit costly is to just lock up channels. You have limitations in lightning. You can't have more than, you can’t have too many pending payments and the amounts of your pending payments is also limited, so you can just lock up channels. You can just send HTLC’s without pre-images and after a while, channels will not accept more pending payments. However, this is not free. So can you tell me why it's not really free to do that? 

```

Also, the use of preimage is inconsistent. There are 6 preimages and 2 pre-images.

**Review by adamjonas — COMMENTED** [review]
A few nits but otherwise good to go!

**Comment by gijswijs** [comment]
Nits fixed. Should be good to go.

File: chaincode-labs/chaincode-residency/2019-06-25-fabrice-drouin-attack-vectors-of-lightning-network.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+There's an ongoing DDos attack  against electrum servers, I don’t know if you have heard of it. They are blacklisting I think 150,000 IPs right now, but it's a battle that is really hard to fight. Something that is really easy to do with lightning but it's a bit costly is to just lock up channels. You have limitations in lightning. You can't have more than, you can’t have too many pending payments and the amounts of your pending payments is also limited, so you can just lock up channels. You can just send HTLC’s without pre-images and after a while channels will not accept more pending payments. However this is not free. So can you tell me why it's not really free to do that? 
+
+Audience Member: You also lock up your own channels.
+
+Fabrice: Yes so it's,...It can be really annoying and if you choose really long routes you can lock up to ten times your funds or even 20 times your funds but it's expensive for you. You need money to do that, but it can be really annoying. 
+
+Audience Member: But there is an asymmetry going on right? It’s pay 1...the benefit is 20 times.
```
[the remark this answers is not in the document]
Comment (adamjonas) [reply chaincode-labs/chaincode-residency/2019-06-25-fabrice-drouin-attack-vectors-of-lightning-network.md]: Sorry. I meant replace throughout. There are still 10 instances of curly quotations.

[the remark this answers is not in the document]
Comment (gijswijs) [reply chaincode-labs/chaincode-residency/2019-06-25-fabrice-drouin-attack-vectors-of-lightning-network.md]: > There are still 10 instances of curly quotations

Yeah, I already replaced them in [this commit](https://github.com/bitcointranscripts/bitcointranscripts/pull/40/commits/95835fcecf75f4c423e5f7fe5e5a7cd415632188) that is already part of this PR.

**Pull request merged**